### PR TITLE
Add unit tests for token validator endpoints

### PR DESCRIPTION
**Pull request by eaudetcobello**
Add unit tests for the ValidateNodeTokenAccessHandler.
Add mock for Provider interface.

### CODE REVIEW

File: src/k8s/pkg/k8sd/api/node_access_handler_test.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+	"testing"
+
+	"github.com/canonical/k8s/pkg/snap"
+	"github.com/canonical/k8s/pkg/snap/mock"
+	. "github.com/onsi/gomega"
+)
+
+func TestValidateNodeTokenAccessHandler(t *testing.T) {
+	const tokenFileName = "tmp-token-file"
+	const tokenHeaderName = "X-Node-Token"
```
Comment (HomayoonAlimohammadi) [inline src/k8s/pkg/k8sd/api/node_access_handler_test.go:17]:
```suggestion
	const (
		tokenFileName   = "tmp-token-file"
		tokenHeaderName = "X-Node-Token"
	)
```

Comment (HomayoonAlimohammadi) [reply src/k8s/pkg/k8sd/api/node_access_handler_test.go:17]: I was actually thinking that we can centralize this `tokenHeaderName` (`node-token`) to be used in both `endpoints.go` and in this test. WDYT?

**Review by HomayoonAlimohammadi — APPROVED** [review]
Great work, thanks a lot @eaudetcobello! LGTM overall. Left some minor suggestions.

File: src/k8s/pkg/k8sd/api/node_access_handler_test.go (hunk 1 above)
Comment (eaudetcobello) [reply src/k8s/pkg/k8sd/api/node_access_handler_test.go:17]: Done

File: src/k8s/pkg/k8sd/api/node_access_handler_test.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+			expectErr:          true,
+			createFile:         true,
+		},
+		{
+			name:               "missing header token",
+			tokenHeaderContent: "",
+			tokenFileContent:   "node-token",
+			expectErr:          true,
+			createFile:         true,
+		},
```
[the remark this answers is not in the document]
Comment (eaudetcobello) [reply src/k8s/pkg/k8sd/api/node_access_handler_test.go:48]: Done

File: src/k8s/pkg/k8sd/api/node_access_handler_test.go [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			g := NewWithT(t)
+			var err error
+			if tc.createFile {
+				err = os.WriteFile(path.Join(tempDir, tokenFileName), []byte(tc.tokenFileContent), 0o644)
+				g.Expect(err).To(BeNil())
```
[the remark this answers is not in the document]
Comment (eaudetcobello) [reply src/k8s/pkg/k8sd/api/node_access_handler_test.go:68]: Done

**Review by bschimke95 — APPROVED** [review]
LGTM, address the linter issue then we can merge.

**Review by HomayoonAlimohammadi — COMMENTED** [review]
Sorry to bring this up after an approval. Maybe we can put the `node-token` header name in the k8s-snap-api repo? Like we have for the `CAPIAuthToken`?
And then also change the usage accordingly in the `cluster-api-k8s` repo?

**Pull request merged**